lib/src/arithmetic/adder.dart

```diff
@@ -21,9 +21,28 @@ abstract class Adder extends Module {
   @protected
   late final Logic b;
 
-  /// The addition results [sum].
+  /// The addition results [out] including carry bit
+  Logic get out => output('out');
+
+  /// The carry results [carryOut].
+  Logic get carryOut => output('carryOut');
+
+  /// The addition results [sum] including carry bit
   Logic get sum => output('sum');
 
+  /// Implementation needs to provide a method for calculating the full sum
```

Comment:
Done for existing adders and base computes the sum.

Reply:
Yet I now see a fundamental problem.  The base class constructor is called before the derived class.  So if the base class asks for CalculateOut()  to build the sum, that is a call to the derived class which has not yet built what it needs to compute the sum.  I have to create a prefix tree in the derived class  for example before I can assign out.  The alternative is I have to now have a Logic to hold the out to break this.

Reply:
My attempts on this all fail.  There is a circularity formed that makes it difficult to decompose the logic into base class and derived class when you look at the order of construction.  late initialization, getters, and and the addOutput() routine need a good design pattern defined for this and I'm just not figuring out the right pattern.  I was using a pattern of inputs only because the problem seems to be on the output side.

Reply:
Perhaps the child classes should be defining their output signal as a `final` variable (prior to call to `super`), then assigning the computed value into that output signal?

Reply:
I think this was the resolution:  use a Logic in the child class to break the cycle and pass back the value.
I think the adder base class is now a reasonable API.